File: backend/jooq-codegen-customization/src/main/java/ch/ergon/lernende/wmtippspiel/backend/WmTippspielGeneratorStrategy.java
package ch.ergon.lernende.wmtippspiel.backend;

import org.jooq.codegen.DefaultGeneratorStrategy;
import org.jooq.codegen.GeneratorStrategy;
import org.jooq.meta.Definition;
import org.jooq.meta.TableDefinition;

public class WmTippspielGeneratorStrategy extends DefaultGeneratorStrategy {

    @Override
    public String getJavaClassName(Definition definition, GeneratorStrategy.Mode mode) {
        var defaultName = super.getJavaClassName(definition, mode);
        if (mode == GeneratorStrategy.Mode.DEFAULT && definition instanceof TableDefinition) {
            return defaultName + "Table";
        } else {
            return defaultName;
        }
    }
}

File: backend/jooq-codegen-customization/src/main/kotlin/ch/ergon/lernende/wmtippspiel/backend/WmTippspielGeneratorStrategy.kt
package ch.ergon.lernende.wmtippspiel.backend

import org.jooq.codegen.DefaultGeneratorStrategy
import org.jooq.codegen.GeneratorStrategy
import org.jooq.codegen.GeneratorStrategy.Mode.DEFAULT
import org.jooq.meta.Definition
import org.jooq.meta.TableDefinition

class WmTippspielGeneratorStrategy : DefaultGeneratorStrategy() {

    override fun getJavaClassName(definition: Definition, mode: GeneratorStrategy.Mode): String {
        val defaultName = super.getJavaClassName(definition, mode)
        return when {
            mode == DEFAULT && definition is TableDefinition -> "${defaultName}Table"
            else                                             -> defaultName
        }
    }

}